diff --git a/qtbase/src/corelib/global/qlogging.cpp b/qtbase/src/corelib/global/qlogging.cpp
index c6d2e26..cb13e36 100644
--- a/qtbase/src/corelib/global/qlogging.cpp
+++ b/qtbase/src/corelib/global/qlogging.cpp
@@ -1506,7 +1506,7 @@ backtraceFramesForLogMessage(int frameCount,
     // use dladdr() instead of backtrace_symbols()
     QString cachedLibrary;
     const char *cachedFname = nullptr;
-    auto decodeFrame = [&](void *addr) -> DecodedFrame {
+    auto decodeFrame = [&](const void *addr) -> DecodedFrame {
         Dl_info info;
         if (!dladdr(addr, &info))
             return {};
@@ -1537,8 +1537,9 @@ backtraceFramesForLogMessage(int frameCount,
     // This code is protected by QMessagePattern::mutex so it is thread safe on all compilers
     static const QRegularExpression rx(QStringLiteral("^(?:[^(]*/)?([^(/]+)\\(([^+]*)(?:[\\+[a-f0-9x]*)?\\) \\[[a-f0-9x]*\\]$"));
 
-    auto decodeFrame = [&](void *&addr) -> DecodedFrame {
-        QScopedPointer<char*, QScopedPointerPodDeleter> strings(backtrace_symbols(&addr, 1));
+    auto decodeFrame = [&](const void *addr) -> DecodedFrame {
+        void *nonConstAddr = const_cast<void *>(addr);
+        QScopedPointer<char*, QScopedPointerPodDeleter> strings(backtrace_symbols(&nonConstAddr, 1));
         QString trace = QString::fromUtf8(strings.data()[0]);
         QRegularExpressionMatch m = rx.match(trace);
         if (!m.hasMatch())
