diff --git a/qtbase/cmake/QtBuildInformation.cmake b/qtbase/cmake/QtBuildInformation.cmake
index 5c44a7e..1ba3a44 100644
--- a/qtbase/cmake/QtBuildInformation.cmake
+++ b/qtbase/cmake/QtBuildInformation.cmake
@@ -1,6 +1,8 @@
 # Copyright (C) 2022 The Qt Company Ltd.
 # SPDX-License-Identifier: BSD-3-Clause
 
+include(FeatureSummary)
+
 function(qt_internal_set_message_log_level out_var)
     # Decide whether output should be verbose or not.
     # Default to verbose (--log-level=STATUS) in a developer-build and
