diff --git a/qtbase/cmake/QtBaseTopLevelHelpers.cmake b/qtbase/cmake/QtBaseTopLevelHelpers.cmake
index 07893b6..4bf72a4 100644
--- a/qtbase/cmake/QtBaseTopLevelHelpers.cmake
+++ b/qtbase/cmake/QtBaseTopLevelHelpers.cmake
@@ -73,6 +73,7 @@ macro(qt_internal_top_level_end)
 
     # Depends on QtBuildInternalsConfig being included, which is the case whenver any repo is
     # configured.
+    include(QtBuildRepoHelpers)
     qt_internal_qt_configure_end()
 endmacro()
 
diff --git a/qtbase/cmake/QtBuildInformation.cmake b/qtbase/cmake/QtBuildInformation.cmake
index 22fcd0e..d4bac1d 100644
--- a/qtbase/cmake/QtBuildInformation.cmake
+++ b/qtbase/cmake/QtBuildInformation.cmake
@@ -125,6 +125,7 @@ from the build directory")
     set(QT_INTERNAL_BUILD_INSTRUCTIONS_SHOWN "TRUE" CACHE STRING "" FORCE)
 
     if(QT_SUPERBUILD)
+        include(QtFindPackageHelpers)
         qt_internal_save_previously_visited_packages()
     endif()
 endfunction()
