diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0247f67e..1deb5d6f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,26 @@ project(Qt
     LANGUAGES CXX C
 )
 
+# Skip libraries that are not dll.a.
+# This does end up using the bundled libs for some libraries, but
+# who cares. Windows sucks.
+if (MINGW AND NOT BUILD_SHARED_LIBS)
+    function(find_library var)
+        if (${var} AND NOT ${var} MATCHES "dll\\.a$")
+            return()
+        endif()
+
+        _find_library(${var} ${ARGN})
+        if (${var})
+            get_filename_component(lib_name "${${var}}" NAME)
+            if (lib_name MATCHES "dll\\.a$")
+                unset(${var} CACHE)
+                set(${var} "${var}-NOTFOUND" CACHE INTERNAL "" FORCE)
+            endif()
+        endif()
+    endfunction()
+endif()
+
 if(UNIX AND NOT ANDROID)
     enable_language(ASM)
 endif()
